Assets/Scripts/ActorMovement.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ActorMovement : MonoBehaviour
+{
+    [SerializeField]
+    Transform[] waypoints;
```

Comment:
Based on what I see below, I'd argue that an array is not the best way to save these waypoints. 

1. A level designer (read: you or John) shouldn't need to care about the order in which the waypoints are set. 
2. Indexing an array like this is very fragile. A mis-click in the editor that swaps two indices can make computron behave in a totally unexpected way without warning.
3. Failing to set a waypoint would cause a crash, and fixing it would most likely require using the debugger and inspecting the code you use to index the array

I'd recommend exposing several fields to the editor, one for each waypoint.
So, instead of an array, the level designer could directly assign:
RestWaypoint
InputWaypoint
OutputWaypoint
RegisterWaypoint

Exposing these fields would make your code more extensible and maintainable. It prevents some errors and makes the rest easier to spot and debug

Reply:
Done. See newest commit.